File: 백준/Bronze/1924. 2007년/2007년.py
# 일 월 화 수 목 금 토 
# 1,2,3,4,5,6, 7 번째
# 요일 반복 --> 일~토 [0 ~ 6] 반복
# 9월 2일을 구하고 싶다 --> 8월까지의 일자를 전부 합산 후 2일을 더한다(자료구조 형식 조심)


a = ['SUN', 'MON', 'TUE', 'WED', 'THU', 'FRI', 'SAT'] 
b = [31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31]

day = 0
m, d = list(map(int, input().split(' ')))

for i in range(m - 1):
    day += b[i] 

day += d
print(a[day % 7])

File: 백준/Bronze/1924. 2007년/2007년.java
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.BufferedReader;

import java.util.StringTokenizer;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine(), " ");

        int month = Integer.parseInt(st.nextToken());
        int day = Integer.parseInt(st.nextToken());
        int[] day_of_month = {31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31};
        String[] day_of_week = {"MON", "TUE", "WED", "THU", "FRI", "SAT", "SUN"};


        // 누적일수 계산
        int total_days = 0;
        for (int i = 0; i < month - 1; i++) {
            total_days += day_of_month[i];
        }
        total_days += day; // 현재 월의 날짜 추가


        // 요일 계산
        // -1은 배열 인덱스로 조정 --> 1월 1일일때 d_o_w[0]이 "MON" 이기 때문
        String res = day_of_week[ (total_days - 1) % 7 ];
        System.out.println(res);




    }
}